Assets/SEE/GraphProviders/GitGraphProvider.cs

```diff
@@ -0,0 +1,282 @@
+
+                foreach (var commit in commits)
+                {
+                    if (commit.Author.When >= oldCommit.Author.When && commit.Author.When <= newCommit.Author.When)
+                    {
+                        foreach (var parent in commit.Parents)
+                        {
+                            var changes = repo.Diff.Compare<Patch>(parent.Tree, commit.Tree);
+
+                            foreach (var change in changes)
+                            {
+                                var filePath = change.Path;
+                                var id = commit.Author.Email;
+
+                                if (!uniqueContributorsPerFile.ContainsKey(filePath))
+                                {
+                                    uniqueContributorsPerFile[filePath] = new HashSet<string>();
+                                }
+
+                                uniqueContributorsPerFile[filePath].Add(id);
+                            }
+                        }
+                    }
+                }
+                foreach (var entry in uniqueContributorsPerFile)
+                {
+                    foreach (var node in graph.Nodes())
+                    {
+                        if (node.ID.Replace('\\', '/') == entry.Key)
+                        {
+                            node.SetInt("Number of Developers", entry.Value.Count);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Calculates the number of times each file was changed for each file changed between two commits and adds it as a metric to <paramref name="graph"/>.
+        /// </summary>
+        /// <param name="graph">an existing graph where to add the metrics</param>
+        /// <param name="vcsPath">the path to the VCS containing the two revisions to be compared</param>
+        /// <param name="oldRevision">the older revision that constitutes the baseline of the comparison</param>
+        /// <param name="newRevision">the newer revision against which the <see cref="oldRevision"/> is to be compared</param>
+        protected void AddCommitFrequencyMetric(Graph graph, string vcsPath, string oldRevision, string newRevision)
+        {
+            using (var repo = new Repository(vcsPath))
+            {
+                var oldCommit = repo.Lookup<Commit>(oldRevision);
+                var newCommit = repo.Lookup<Commit>(newRevision);
+
+                var commitsBetween = repo.Commits.QueryBy(new CommitFilter
+                {
+                    IncludeReachableFrom = newCommit,
+                    ExcludeReachableFrom = oldCommit
+                });
+
+                Dictionary<string, int> fileCommitCounts = new Dictionary<string, int>();
+
+                foreach (var commit in commitsBetween)
+                {
+                    foreach (var parent in commit.Parents)
```

Comment:
[Do not use `var`.](https://github.com/uni-bremen-agst/SEE/wiki/ReviewChecklist)